xonsh/aliases.py

```diff
@@ -412,9 +412,9 @@ def which(args, stdin=None, stdout=None, stderr=None):
         for abs_name, from_where in matches:
             if ON_WINDOWS:
                 # Use list dir to get correct case for the filename
-                # i.e. windows is case insesitive but case preserving
+                # i.e. windows is case insensitive but case preserving
                 p, f = os.path.split(abs_name)
-                f = next(s for s in os.listdir(p) if s.lower() == f.lower())
+                f = next(s.name for s in scandir(p) if s.name.lower() == f.lower())
```

Comment:
It would be nice to store an `flower = f.lower()` variable once here, if you feel like making that change.